src/api/controllers/EmployersController.ts

```diff
@@ -12,46 +14,96 @@ import { RoutedControllerBase } from "./RoutedControllerBase";
 class EmployersController extends RoutedControllerBase {
 	public static readonly SUBPATH: string = "employers";
 
-	private static readonly MAX_GETEMPLOYERS_DETAILED: number = 100;
-
 	private static readonly MAX_GETEMPLOYERS_STANDARD: number = 500;
 
 	private readonly recordLoader: EmployerRecordLoader = new EmployerRecordLoader(process.env.RAZZLE_PUBLIC_DIR || "/", "employers");
 
-	private employers: EmployerRecord[] = [];
+	private employers: { [id: string]: EmployerRecord } = {};
+
+	private hasEmployers: boolean = false;
+
+	public async getEmployer(req: express.Request, res: express.Response): Promise<void> {
+		const id: string | null = req.params.employerId || null;
+
+		if (!id) {
+			res.status(HttpStatusCodes.BAD_REQUEST).send();
+
+			return;
+		}
+
+		let employerRecord: EmployerRecord | null = null;
 
-	public async getEmployersList(req: express.Request, res: express.Response): Promise<void> {
-		if (!req.header(HttpRequestHeaders.IF_NONE_MATCH) || this.employers.length === 0) {
-			this.employers = await this.recordLoader.loadAllAsync();
+		if (req.header(HttpRequestHeaders.IF_NONE_MATCH) && id in this.employers) {
+			employerRecord = this.employers[id];
+		} else if (await this.recordLoader.existsAsync(id)) {
+			employerRecord = await this.recordLoader.loadAsync(id);
 		}
 
-		// TODO Allow individual field selection as well.
-		const isDetailed: boolean = req.query.$select === "*";
+		if (!employerRecord) {
+			res.status(HttpStatusCodes.NOT_FOUND).send();
 
-		const max: number =
-			isDetailed ? EmployersController.MAX_GETEMPLOYERS_DETAILED : EmployersController.MAX_GETEMPLOYERS_STANDARD;
+			return;
+		}
 
-		const skipParam: string = req.query.$skip || "0";
-		const topParam: string = req.query.$top || `${max}`;
+		this.employers[id] = employerRecord;
 
-		const skipNumber: number = parseInt(skipParam, 10);
-		const topNumber: number = parseInt(topParam, 10);
+		res.setHeader(HttpResponseHeaders.CONTENT_TYPE, MimeTypes.APPLICATION_JSON);
+		res.send(employerRecord);
+	}
 
-		const employers: EmployerRecord[] =
-			isDetailed
-				? this.employers
-				: this.employers.map((e: EmployerRecord) => EmployerRecord.shallowClone(e, false));
+	public async getEmployerCategories(req: express.Request, res: express.Response): Promise<void> {
+		if (!req.header(HttpRequestHeaders.IF_NONE_MATCH) || !this.hasEmployers) {
+			await this.reloadEmployersAsync();
+		}
 
-		const returnedEmployers: EmployerRecord[] = employers.slice(skipNumber, skipNumber + topNumber);
+		const categories: string[] = [ "technology", "healthcare", "grocery" ];
 
+		/*
+		for (const employer of Object.values(this.employers)) {
```

Comment:
Zombie code?

Reply:
Enabling next PR with home page and category support. Leaving here for now as it doesn't hurt anything.